ldpd/lde.c

```diff
@@ -1709,6 +1722,32 @@ on_get_label_chunk_response(uint32_t start, uint32_t end)
 		current_label_chunk = listtail(label_chunk_list);
 }
 
+void
+lde_free_label(uint32_t label)
+{
+	struct listnode	*node;
+	struct label_chunk	*label_chunk;
+	uint64_t		pos;
+
+	for (ALL_LIST_ELEMENTS_RO(label_chunk_list, node, label_chunk)) {
```

Comment:
Hmm - I know @rwestphal asked for the `_RO` version of the iterator, but ... we do want to delete the list element if we successfully free/release it, so I don't think that's the correct flavor?

Reply:
We should not use ALL_LIST_ELEMENTS_RO only if we are continuing iteration after deleting a node, because its going call listnextnode(node) inside macro.
But here, we do break immediately after deleting a node. So, I hope its OK.